src/units.cpp

```diff
@@ -464,6 +464,91 @@ double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
     return 0.0;
 }
 
+using unitsMap = std::map<std::string, double>;
+
+void updateUnitMap(const UnitsPtr &units, unitsMap &unitMap, double exp = 1.0)
+{
+    if (units->isBaseUnit()) {
+        auto found = unitMap.find(units->name());
+        if (found == unitMap.end()) {
+            unitMap.emplace(units->name(), 1.0);
+        } else {
+            auto unit = unitMap.find(units->name());
+            unit->second += exp;
```

Comment:
`found->second += exp;` ?

Reply:
Ah yes, that's a bit less redundant than what I had before.